src/network/clientpackethandler.cpp

```diff
@@ -404,28 +404,21 @@ void Client::handleCommand_ChatMessage(NetworkPacket *pkt)
 		wstring message
 	 */
 
-	ChatMessage *chatMessage = new ChatMessage();
+	auto chatMessage = std::unique_ptr<ChatMessage>(new ChatMessage());
 	u8 version, message_type;
 	*pkt >> version >> message_type;
 
-	if (version != 1 || message_type >= CHATMESSAGE_TYPE_MAX) {
-		delete chatMessage;
+	if (version != 1 || message_type >= CHATMESSAGE_TYPE_MAX)
 		return;
-	}
 
 	u64 timestamp;
 	*pkt >> chatMessage->sender >> chatMessage->message >> timestamp;
 	chatMessage->timestamp = static_cast<std::time_t>(timestamp);
 
 	chatMessage->type = (ChatMessageType) message_type;
 
-	// @TODO send this to CSM using ChatMessage object
-	if (modsLoaded() && m_script->on_receiving_message(
-			wide_to_utf8(chatMessage->message))) {
-		// Message was consumed by CSM and should not be handled by client
-		delete chatMessage;
-	} else {
-		pushToChatQueue(chatMessage);
+	if (!modsLoaded() || !m_script->on_receiving_message(chatMessage.get())) {
+		pushToChatQueue(chatMessage.release());
```

Comment:
What's the point of the unique_ptr when you release it in the same function?

Reply:
I was going to move it into the chat queue as well, but was unsure if that would be too much of a change

Reply:
Well, there is a big one - if an exception is thrown, say by a CSM mod, then there won't be a memory leak